services/navigation/builtin/builtin.go

```diff
@@ -529,7 +529,10 @@ func (svc *builtIn) moveToWaypoint(ctx context.Context, wp navigation.Waypoint,
 	cancelCtx, cancelFn := context.WithCancel(ctx)
 	defer cancelFn()
 	executionID, err := svc.motionService.MoveOnGlobe(cancelCtx, req)
-	if err != nil {
+	if errors.Is(err, motion.ErrGoalWithinPlanDeviation) {
+		// make an exception for the error that is raised when motion is not possible because already at goal.
+		return svc.waypointReached(cancelCtx)
```

Comment:
Should we add a test case for this in the nav tests?

Reply:
Yes, I wasn't sure how to test for this but I can figure something out

Reply:
Added a test